src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -916,19 +911,17 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
                                                     args_modifier);
   }
   *port_num = -1;
-  absl::StatusOr<std::vector<grpc_resolved_address>> resolved_or;
   std::vector<grpc_error_handle> error_list;
   std::string parsed_addr = URI::PercentDecode(addr);
-  absl::string_view parsed_addr_unprefixed{parsed_addr};
   // Using lambda to avoid use of goto.
   grpc_error_handle error = [&]() {
     grpc_error_handle error = GRPC_ERROR_NONE;
-    if (absl::ConsumePrefix(&parsed_addr_unprefixed, kUnixUriPrefix)) {
-      resolved_or = grpc_resolve_unix_domain_address(parsed_addr_unprefixed);
-    } else if (absl::ConsumePrefix(&parsed_addr_unprefixed,
-                                   kUnixAbstractUriPrefix)) {
-      resolved_or =
-          grpc_resolve_unix_abstract_domain_address(parsed_addr_unprefixed);
+    auto uri = URI::Parse(parsed_addr);
+    const auto& address_parser_registry =
+        CoreConfiguration::Get().address_parser_registry();
+    absl::StatusOr<std::vector<grpc_resolved_address>> resolved_or;
+    if (uri.ok() && address_parser_registry.HasScheme(uri->scheme())) {
+      resolved_or = address_parser_registry.Parse(*uri);
```

Comment:
It looks to me like this change actually adds support for a single URI supporting multiple addresses for the `ipv4`, `ipv6`, `unix`, and `unix-abstract` schemes, which was not previously supported on the server side.  I'd prefer not to add that support.

IMHO, it was a mistake for our server-side APIs to support DNS names in the first place, and that's currently the only case where the server will actually bind to multiple addresses for a single URI.  I don't think we should support that for any new cases here.

Reply:
Removed the support for comma separated addresses.

We do not support DNS names now, nor have we ever. We do support resolution via getaddrinfo, which means "please read from /etc/hosts". IMHO the biggest mistake here was to place that API under the DNS abstraction.